2013-11-25 17:50:34.7057|Debug|e780cc2c-f89f-4c94-b876-4e9c9f4479f9|Image - Download: e951e548-ed5b-43ec-8253-64540fb4a639 small 
2013-11-25 17:50:34.7057|Debug|657dadc7-19ea-44e7-b949-a2de32e1befb|Image - Download: 5fe80f9e-d6d6-4043-86da-749cd2aa00d9 small 
2013-11-25 17:50:34.8917|Debug|dd25341a-39d8-4415-acea-f6363798c276|Image - Download: 7c13172a-2c43-4508-b411-a9557fb3702f small 
2013-11-25 17:50:34.9287|Debug|ffd33c71-139c-4181-b8c0-04b4879e6b71|Image - Download: 64097422-f530-40c8-8762-48b9237e7ac1 small 
2013-11-25 17:50:34.9437|Debug|9d8d501b-8964-44bc-af4b-4e33c69e6522|Image - Download: e50e2ca8-f828-4ef6-b440-e75f5d18ac4c small 
2013-11-25 17:50:34.9437|Debug|0c95262f-a224-4765-a7a6-92b178f6ca52|Image - Download: e750923b-5665-4bc2-96bd-081baff08d61 small 
2013-11-25 17:50:34.9587|Debug|86e33fc1-d0de-4f3f-86bc-7219baa3304b|Image - Download: d6dea20a-a300-4fa4-9607-0779bdb568c9 small 
2013-11-25 17:50:34.9907|Debug|a9cde279-27c3-4654-9087-4d845a94e8c6|Image - Download: b7885c90-8a96-470a-a4bb-c900ceb1dee7 small 
2013-11-25 17:50:35.0717|Debug|c75076c2-61f4-442b-893b-4ba69a8abe47|Image - Download: a751845c-22de-4a29-a8f2-cf406568ae7c small 
2013-11-25 17:50:35.1197|Debug|4775db0f-21c7-4ca3-8224-5370aab55326|Image - Download: 9cb5190d-8aa4-4be9-8764-cd228345a50f small 
2013-11-25 17:50:40.3400|Debug|26cdabe9-9e14-4738-aa92-ae8a39c8a7da|Image - Download: cecf6fc2-844d-42fc-bab7-53bdb20aea0b large 
2013-11-25 17:50:40.3400|Debug|7734e847-684f-4fc8-87b0-6007b3b7a162|Image - Download: e750923b-5665-4bc2-96bd-081baff08d61 large 
2013-11-25 17:50:40.3400|Debug|9bb35220-c16d-48d9-aaed-4a39e2b94c11|Image - Download: e750923b-5665-4bc2-96bd-081baff08d61 thumbnail 
2013-11-25 17:50:40.3400|Debug|ab48e62e-056a-4f00-b33a-0a3dd613f3a3|Image - Download: 9b92a3ee-78cb-4310-a327-c30f99800c88 thumbnail 
2013-11-25 17:50:44.2022|Debug|d7dcb971-5326-460e-8a3d-3d92f3e5638a|Image - Download: 7c13172a-2c43-4508-b411-a9557fb3702f small 
2013-11-25 17:50:44.2022|Debug|b2489a93-4da4-4960-a42e-20ac9849940d|Image - Download: 64097422-f530-40c8-8762-48b9237e7ac1 small 
2013-11-25 17:50:44.2102|Debug|49dd57be-272c-476e-994f-9548bee2a0db|Image - Download: 9cb5190d-8aa4-4be9-8764-cd228345a50f small 
2013-11-25 17:50:44.2102|Debug|d0a43fa2-74da-4f9f-90ef-b5cae0a70701|Image - Download: 5fe80f9e-d6d6-4043-86da-749cd2aa00d9 small 
2013-11-25 17:50:44.2102|Debug|a300717f-936f-465e-b4aa-614a8f7e7abb|Image - Download: e50e2ca8-f828-4ef6-b440-e75f5d18ac4c small 
2013-11-25 17:50:44.2282|Debug|5c381cef-984e-4e10-8777-7b166dc5b619|Image - Download: e951e548-ed5b-43ec-8253-64540fb4a639 small 
2013-11-25 17:50:44.2282|Debug|aaed1bd5-c5ad-4050-8a8d-5fa90bd4e606|Image - Download: e750923b-5665-4bc2-96bd-081baff08d61 small 
2013-11-25 17:50:44.2452|Debug|5816aa97-4b41-4bb8-a6cb-d29847905b56|Image - Download: a751845c-22de-4a29-a8f2-cf406568ae7c small 
2013-11-25 17:50:44.2452|Debug|dbf855a1-22fd-4ef1-a15a-04e6e34a2df8|Image - Download: d6dea20a-a300-4fa4-9607-0779bdb568c9 small 
2013-11-25 17:50:44.2452|Debug|b23ef81c-55bd-4fc6-832f-27633d50d886|Image - Download: b7885c90-8a96-470a-a4bb-c900ceb1dee7 small 
2013-11-25 17:50:49.1095|Debug|a206d48b-5f41-4590-92d0-0089395bdd75|Image - Download: d6dea20a-a300-4fa4-9607-0779bdb568c9 large 
2013-11-25 17:50:49.1095|Debug|d75ca03d-2eae-4f31-aedc-4550a1b46653|Image - Download: d6dea20a-a300-4fa4-9607-0779bdb568c9 thumbnail 
2013-11-25 17:50:49.1095|Debug|1d8a862f-189f-4482-ade5-9b2547073550|Image - Download: 316410f9-cec4-4adf-8554-f49ee0a17f04 thumbnail 
2013-11-25 17:50:49.1095|Debug|20943f7f-389e-4ffa-a4c0-f27ff2a0f299|Image - Download: a9111a80-e8f3-4a8e-9998-6b84d7e6ccd5 thumbnail 
2013-11-25 17:50:49.1605|Debug|487651a0-7e3a-4904-8d6a-dba8db6db05a|Image - Download: b104466c-0070-4cbf-ba97-e2a6c978c8bc large 
2013-11-25 17:50:56.8970|Debug|b6345456-793a-419a-96cd-e2fd9c04c01b|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 thumbnail 
2013-11-25 17:50:56.8970|Debug|4c865736-5d05-4510-8bd2-bbd227e51fa5|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 large 
2013-11-25 17:51:41.2555|Debug|1b3d6a31-4195-4dff-9af1-332377a978d3|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 large 
2013-11-25 17:51:41.2615|Debug|2d975818-53a0-48b2-846a-a0b633dc2160|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 thumbnail 
2013-11-25 17:52:11.7512|Debug|c7471811-f5c1-4f32-b44c-86424d2a4617|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 thumbnail 
2013-11-25 17:52:11.7512|Debug|97e3553d-a7e4-4c07-bd4c-135ef54075d6|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 large 
2013-11-25 17:52:13.8904|Debug|088ed0cc-cece-46c3-a456-475034c71a4b|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 thumbnail 
2013-11-25 17:52:13.8964|Debug|6c34f954-47c2-47dc-a5a4-90239ea4d339|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 large 
2013-11-25 17:52:15.8395|Debug|692dd099-2874-4a68-b41b-c45f9b99ce8d|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 thumbnail 
2013-11-25 17:52:15.8395|Debug|05166958-b639-4101-960d-e85acb56ab88|Image - Download: d49358db-86f2-470f-a4ca-d8951aade9e8 large 
2013-11-25 18:15:26.7180|Debug|31ac6aa4-e4a1-46fc-bb33-38074bd2987f|Image - Download: e951e548-ed5b-43ec-8253-64540fb4a639 thumbnail 
2013-11-25 18:15:26.7180|Debug|412bdf77-a95b-4cc9-b85d-b3e7af9296a8|Image - Download: e50e2ca8-f828-4ef6-b440-e75f5d18ac4c thumbnail 
2013-11-25 18:15:26.7790|Debug|bc210307-cdf4-4d6e-9cb9-4ececbcc2905|Image - Download: 5fe80f9e-d6d6-4043-86da-749cd2aa00d9 thumbnail 
2013-11-25 18:15:26.8450|Debug|00d0ff2d-fe54-4092-9407-ebac9fd7fef2|Image - Download: 7c13172a-2c43-4508-b411-a9557fb3702f thumbnail 
2013-11-25 18:15:26.8450|Debug|9a262c5b-d8d8-43b6-b5de-7e43e379e646|Image - Download: d6dea20a-a300-4fa4-9607-0779bdb568c9 thumbnail 
2013-11-25 18:15:26.8450|Debug|fe48198c-ddba-42f3-a24a-98389c3d4593|Image - Download: e750923b-5665-4bc2-96bd-081baff08d61 thumbnail 
2013-11-25 18:15:26.8590|Debug|51f969ed-ab83-4916-91bc-06cc473f3347|Image - Download: b7885c90-8a96-470a-a4bb-c900ceb1dee7 thumbnail 
2013-11-25 18:15:26.8590|Debug|7c001503-5952-414f-91a7-4a5b3174d305|Image - Download: 64097422-f530-40c8-8762-48b9237e7ac1 thumbnail 
2013-11-25 18:15:26.8590|Debug|f3d86d2d-ab3a-4e08-a236-598a2185ef9d|Image - Download: 9cb5190d-8aa4-4be9-8764-cd228345a50f thumbnail 
2013-11-25 18:15:26.9000|Debug|733e68b7-2362-426f-9d76-4cc751e3e050|Image - Download: a751845c-22de-4a29-a8f2-cf406568ae7c thumbnail 
2013-11-25 18:31:30.9062|Error|b2bc8aa6-ab3e-4b0f-8fab-be43c6f5b277|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:31.7122|Error|aca5db0c-5f31-4b1d-a6a3-0c322efd477f|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:32.1292|Error|e27a1dc4-6455-422a-9970-ecb6a19786a4|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:32.2843|Error|06dac4c1-1af5-451d-8d3e-331b05485016|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:32.3763|Error|a3e2187c-f9eb-4968-a46d-35246157738e|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:32.5783|Error|8b6bb2fc-ad9b-4f99-be2b-1d6dc651a9a7|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:33.0923|Error|c3329544-d646-4e86-8b5e-768856b2274e|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:35.8675|Error|aca5db0c-5f31-4b1d-a6a3-0c322efd477f|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:31:36.4045|Error|aca5db0c-5f31-4b1d-a6a3-0c322efd477f|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:31:36.8605|Error|06dac4c1-1af5-451d-8d3e-331b05485016|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:31:36.8605|Error|06dac4c1-1af5-451d-8d3e-331b05485016|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:31:37.9466|Error|8b6bb2fc-ad9b-4f99-be2b-1d6dc651a9a7|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:31:37.9466|Error|8b6bb2fc-ad9b-4f99-be2b-1d6dc651a9a7|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:31:37.9736|Error|e27a1dc4-6455-422a-9970-ecb6a19786a4|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:31:37.9736|Error|e27a1dc4-6455-422a-9970-ecb6a19786a4|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:31:37.9826|Error|a3e2187c-f9eb-4968-a46d-35246157738e|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:31:37.9826|Error|a3e2187c-f9eb-4968-a46d-35246157738e|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:31:38.5146|Error|b5b0c74d-1970-48f1-8ebc-cfe741fd2fe3|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:31:38.5626|Error|a3e2187c-f9eb-4968-a46d-35246157738e|System.Data.EntityCommandExecutionException:An error occurred while executing the command definition. See the inner exception for details.
   at System.Data.EntityClient.EntityCommandDefinition.ExecuteStoreCommands(EntityCommand entityCommand, CommandBehavior behavior)
   at System.Data.Objects.Internal.ObjectQueryExecutionPlan.Execute[TResultType](ObjectContext context, ObjectParameterCollection parameterValues)
   at System.Data.Objects.ObjectQuery`1.GetResults(Nullable`1 forMergeOption)
   at System.Data.Objects.ObjectQuery`1.System.Collections.Generic.IEnumerable<T>.GetEnumerator()
   at System.Data.Entity.Internal.Linq.InternalQuery`1.GetEnumerator()
   at System.Data.Entity.Infrastructure.DbQuery`1.System.Collections.Generic.IEnumerable<TResult>.GetEnumerator()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at Commerce.Application.Email.AdminEmailBuilder.MessageFactory() in C:\DEV\Pleiades\Commerce.Domain\Email\AdminEmailBuilder.cs:line 34
   at Commerce.Application.Email.AdminEmailBuilder.SystemError(Guid activityId, Exception exception) in C:\DEV\Pleiades\Commerce.Domain\Email\AdminEmailBuilder.cs:line 70
   at Commerce.Web.Plumbing.ErrorNotification.Send(Exception ex) in C:\DEV\Pleiades\Commerce.Web\Plumbing\ErrorNotification.cs:line 19

INNER EXCEPTION
System.InvalidOperationException:There is already an open DataReader associated with this Command which must be closed first.
   at System.Data.SqlClient.SqlInternalConnectionTds.ValidateConnectionForExecute(SqlCommand command)
   at System.Data.SqlClient.SqlConnection.ValidateConnectionForExecute(String method, SqlCommand command)
   at System.Data.SqlClient.SqlCommand.ValidateCommand(String method, Boolean async)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, DbAsyncResult result)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.EntityClient.EntityCommandDefinition.ExecuteStoreCommands(EntityCommand entityCommand, CommandBehavior behavior) 
2013-11-25 18:31:38.5726|Error|e27a1dc4-6455-422a-9970-ecb6a19786a4|System.Data.EntityCommandExecutionException:An error occurred while executing the command definition. See the inner exception for details.
   at System.Data.EntityClient.EntityCommandDefinition.ExecuteStoreCommands(EntityCommand entityCommand, CommandBehavior behavior)
   at System.Data.Objects.Internal.ObjectQueryExecutionPlan.Execute[TResultType](ObjectContext context, ObjectParameterCollection parameterValues)
   at System.Data.Objects.ObjectQuery`1.GetResults(Nullable`1 forMergeOption)
   at System.Data.Objects.ObjectQuery`1.System.Collections.Generic.IEnumerable<T>.GetEnumerator()
   at System.Data.Entity.Internal.Linq.InternalQuery`1.GetEnumerator()
   at System.Data.Entity.Infrastructure.DbQuery`1.System.Collections.Generic.IEnumerable<TResult>.GetEnumerator()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at Commerce.Application.Email.AdminEmailBuilder.MessageFactory() in C:\DEV\Pleiades\Commerce.Domain\Email\AdminEmailBuilder.cs:line 34
   at Commerce.Application.Email.AdminEmailBuilder.SystemError(Guid activityId, Exception exception) in C:\DEV\Pleiades\Commerce.Domain\Email\AdminEmailBuilder.cs:line 70
   at Commerce.Web.Plumbing.ErrorNotification.Send(Exception ex) in C:\DEV\Pleiades\Commerce.Web\Plumbing\ErrorNotification.cs:line 19

INNER EXCEPTION
System.InvalidOperationException:There is already an open DataReader associated with this Command which must be closed first.
   at System.Data.SqlClient.SqlInternalConnectionTds.ValidateConnectionForExecute(SqlCommand command)
   at System.Data.SqlClient.SqlConnection.ValidateConnectionForExecute(String method, SqlCommand command)
   at System.Data.SqlClient.SqlCommand.ValidateCommand(String method, Boolean async)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, DbAsyncResult result)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.EntityClient.EntityCommandDefinition.ExecuteStoreCommands(EntityCommand entityCommand, CommandBehavior behavior) 
2013-11-25 18:32:34.0928|Error|e80eeeeb-e5f0-472c-8c8e-636914beed22|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:35.0568|Error|16f3daf1-f57d-4bf4-a51b-6ce8be7e02d3|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:35.2549|Error|099d46b7-6755-49db-a7f3-b3b9d231554e|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:35.7349|Error|c5510346-6052-43dc-a76b-7eb772d4d03b|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:36.1449|Error|f37e63df-6730-4634-893c-3e34b7c22268|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:36.3089|Error|d2a15e06-989a-4689-bd0c-ec1a9443e4f2|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:37.5420|Error|16f3daf1-f57d-4bf4-a51b-6ce8be7e02d3|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:32:37.5800|Error|16f3daf1-f57d-4bf4-a51b-6ce8be7e02d3|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:32:37.6860|Error|099d46b7-6755-49db-a7f3-b3b9d231554e|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:32:37.6860|Error|099d46b7-6755-49db-a7f3-b3b9d231554e|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:32:38.9821|Error|f2fe16ea-0a6d-4371-a860-61e03b232441|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:39.2831|Error|814654d5-7ef0-460e-9388-3ce4548dfddf|System.Net.WebException:The remote name could not be resolved: 'api.stripe.com'
   at System.Net.HttpWebRequest.GetResponse()
   at Stripe.Requestor.ExecuteWebRequest(WebRequest webRequest)
   at Stripe.Requestor.PostString(String url, String apiKey)
   at Stripe.StripeChargeService.Refund(String chargeId, Nullable`1 refundAmountInCents, Nullable`1 refundApplicationFee)
   at Commerce.Application.Payment.StripePaymentProcessor.Refund(Transaction originalTransaction, Decimal amount) in C:\DEV\Pleiades\Commerce.Domain\Payment\StripePaymentProcessor.cs:line 60 
2013-11-25 18:32:39.8561|Error|f37e63df-6730-4634-893c-3e34b7c22268|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:32:39.8561|Error|f37e63df-6730-4634-893c-3e34b7c22268|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:32:40.7022|Error|c5510346-6052-43dc-a76b-7eb772d4d03b|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:32:40.7022|Error|c5510346-6052-43dc-a76b-7eb772d4d03b|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
2013-11-25 18:32:41.8362|Error|814654d5-7ef0-460e-9388-3ce4548dfddf|URL:http://localhost/Commerce.Web/Admin/ManageOrder/Refund?externalId=ZKU-18731&orderLineIds=3 - IsAjaxRequest: True 
2013-11-25 18:32:41.8362|Error|814654d5-7ef0-460e-9388-3ce4548dfddf|System.Data.Entity.Infrastructure.DbUpdateException:An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at Commerce.Web.Areas.Admin.Controllers.ManageOrderController.Refund(String externalId, String orderLineIds) in C:\DEV\Pleiades\Commerce.Web\Areas\Admin\Controllers\ManageOrderController.cs:line 53
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41()
   at System.Web.Mvc.Async.AsyncResultWrapper.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<>c__DisplayClass2a.<BeginInvokeAction>b__20()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult)

INNER EXCEPTION
System.Data.OptimisticConcurrencyException:Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. Refresh ObjectStateManager entries.
   at System.Data.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   at System.Data.EntityClient.EntityAdapter.Update(IEntityStateManager entityCache)
   at System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges() 
